### ISSUE TITLE
Parallel failure with --enable-complex in adaptivity_ex3

### ISSUE THREAD

**Issue by roystgnr** [libMesh/libmesh#1495]
In the dimension=1 refinement_type=h case, in dbg and devel modes, when --enable-complex is configured and I run in parallel, I get a failure reported by MPICH2:

```
Assertion `libmesh_mpi_error_code == MPI_SUCCESS' failed.
libmesh_mpi_error_code = 875636844
MPI_SUCCESS = 0
Message truncated, error stack:
MPI_Allreduce(907)......................: MPI_Allreduce(sbuf=0x7fffe501648c, rbuf=0x7fffe5016488, count=1, MPI_UNSIGNED, MPI_MIN, MPI_COMM_WORLD) failed
MPIR_Allreduce_impl(764)................:
MPIR_Allreduce_intra(238)...............:
MPIR_Reduce_impl(1070)..................:
MPIR_Reduce_intra(878)..................:
MPIR_Reduce_binomial(186)...............:
MPIDI_CH3_PktHandler_EagerShortSend(363): Message from rank 1 and tag 11 truncated; 8 bytes received but buffer
```

In opt mode the run just hangs.

This failure apparently has existed ever since that test was added in 1a8f034; I haven't ever been running the --enable-complex builds in parallel until now, and we don't have it in CI.

What's especially astonishing is that the dimension=1 test doesn't run and fail until *after* the 2D tests have already run and passed...

The error seems to be hitting when we're in the middle of deleting temporary PetscVectors used by ErrorVector::plot_error.

**Comment by roystgnr** [comment]
I think I've found the bug.  @friedmud fixed the bad parallelism test in the ```PetscVector<Real>::localize_to_one()``` specialization all the way back in #190, but none of us ever noticed the exact same silly bug in the Complex specialization.

It only affects one-element or mis-partitioned meshes with --enable-complex users, so IMHO the fix won't be worth backporting despite the bug breaking parallel "make check", but I'll at least get it fixed in master in now.  Afterward we'll probably want to start running our --enable-complex tests with more processors, just to be paranoid.  Using 2 or 4 ought to be cheap enough since we only do the libMesh tests for those, not the MOOSE build and tests.

**Comment by jwpeterson** [comment]
> Afterward we'll probably want to start running our --enable-complex tests with more processors, just to be paranoid.

We still have the issue on our end that there is no complex-enabled PETSc on the build boxes, therefore we fall back on Eigen in --enable-complex mode, therefore we can only run tests in serial. If you really want this I'll have to convince @milljm that we need to add complex-enabled PETSc to the MOOSE package.

**Comment by roystgnr** [comment]
Shoot, I forgot about your modules issue.

I do really want this, but not enough to make more work for others.  I'll see what I can set up on my end in the medium term.

**Comment by roystgnr** [comment]
This particular issue was fixed by #1496; the problem of "we have an O(2^N) explosion of features and only O(1) CI build box time to test them with" is a separate issue.
